network/controller/bootstrap/authorization.go

```diff
@@ -0,0 +1,173 @@
+	OpRejected
+)
+
+// AuthorizationRequest
+type AuthorizationRequest struct {
+	Certificate core.Certificate
+}
+
+// AuthorizationResponse
+type AuthorizationResponse struct {
+	Code      OperationCode
+	Error     string
+	SessionID SessionID
+}
+
+// RegistrationRequest
+type RegistrationRequest struct {
+	SessionID SessionID
+	JoinClaim *packets.NodeJoinClaim
+}
+
+// RegistrationResponse
+type RegistrationResponse struct {
+	Code  OperationCode
+	Error string
+}
+
+func init() {
+	gob.Register(&AuthorizationRequest{})
+	gob.Register(&AuthorizationResponse{})
+	gob.Register(&RegistrationRequest{})
+	gob.Register(&RegistrationResponse{})
+}
+
+// Authorize node on the discovery node (step 2 of the bootstrap process)
+func (ac *AuthorizationController) Authorize(ctx context.Context, discoveryNode *DiscoveryNode, certificate core.Certificate) (SessionID, error) {
+	inslogger.FromContext(ctx).Infof("Authorizing on host: %s", discoveryNode)
+
+	request := ac.transport.NewRequestBuilder().Type(types.Authorize).Data(&AuthorizationRequest{
+		Certificate: certificate,
+	}).Build()
+	future, err := ac.transport.SendRequestPacket(request, discoveryNode.Host)
+	if err != nil {
+		return 0, errors.Wrapf(err, "Error sending authorize request")
+	}
+	response, err := future.GetResponse(ac.options.PacketTimeout)
+	if err != nil {
+		return 0, errors.Wrapf(err, "Error getting response for authorize request")
+	}
+	data := response.GetData().(*AuthorizationResponse)
+	if data.Code == OpRejected {
+		return 0, errors.New("Authorize rejected: " + data.Error)
+	}
+	return data.SessionID, nil
+}
+
+// Register node on the discovery node (step 4 of the bootstrap process)
+func (ac *AuthorizationController) Register(ctx context.Context, discoveryNode *DiscoveryNode, sessionID SessionID) error {
+	inslogger.FromContext(ctx).Infof("Registering on host: %s", discoveryNode)
+
+	request := ac.transport.NewRequestBuilder().Type(types.Register).Data(&RegistrationRequest{
+		SessionID: sessionID,
+		JoinClaim: ac.keeper.GetOriginClaim(),
+	}).Build()
+	future, err := ac.transport.SendRequestPacket(request, discoveryNode.Host)
+	if err != nil {
+		return errors.Wrapf(err, "Error sending register request")
+	}
+	response, err := future.GetResponse(ac.options.PacketTimeout)
+	if err != nil {
+		return errors.Wrapf(err, "Error getting response for register request")
+	}
+	data := response.GetData().(*RegistrationResponse)
+	if data.Code == OpRejected {
+		return errors.New("Register rejected: " + data.Error)
+	}
+	return nil
+}
+
+func (ac *AuthorizationController) checkClaim(sessionID SessionID, claim *packets.NodeJoinClaim) error {
+	session, err := ac.sessionManager.ReleaseSession(sessionID)
+	if err != nil {
+		return errors.Wrapf(err, "Error getting section %d for authorization", sessionID)
+	}
+	if !claim.NodeRef.Equal(session.NodeID) {
+		return errors.New("Claim node ID is not equal to session node ID")
+	}
+	// TODO: check claim signature
```

Comment:
Let's check it.

Reply:
Let's create a separate task for signing and checking the signature of join claim

Reply:
Done